core/src/main/scala/pipeflow/system/actors/PeriodicSchedulerActor.scala

```diff
@@ -0,0 +1,89 @@
+package pipeflow.system.actors
+
+import java.time.{Clock, OffsetDateTime}
+import java.util.concurrent.TimeUnit
+
+import akka.actor.{Actor, ActorRef, PoisonPill, Props}
+import akka.event.LoggingReceive
+import akka.pattern.pipe
+import pipeflow.dsl.nodes.Node
+import pipeflow.system.PipeFlowSystem.NodeBuilder
+import pipeflow.system.iso8601.{IntervalDuration, RepeatingInterval}
+
+import scala.concurrent.duration.FiniteDuration
+import scala.concurrent.{ExecutionContext, Future}
+
+object PeriodicSchedulerActor {
+
+  case class Tick(recurrences: Option[Long], scheduledDateTime: OffsetDateTime)
+
+  case class NodeCreated(node: Node)
```

Comment:
you can make this a value type to avoid allocation, `case class NodeCreated(node: Node) extends AnyVal`